src/EventExport/EventExportService.php

```diff
@@ -115,6 +127,10 @@ public function exportEvents(
             throw ($e);
         }
 
+        if ($totalItemCount > $this->maxAmountOfItems) {
+            throw new MaximumNumberOfExportItemsExceeded();
```

Comment:
Wouldn't this also throw if the `$totalItemCount` is more than the max amount, but there is a selection which is smaller? Eg. total count is 5000, max count is 2000, but selection is 10. In that case it shouldn't throw because we're just exporting 10 items. This whole "pre-query" thing actually seems unnecessary even when we're exporting a selection.